[Sitebars endpoint] Add tests for slashing behavior

adamziel opened the pull request:
## Description
This PR adds coverage for the correct usage of `wp_slash` in `update_item` method as per  https://github.com/WordPress/gutenberg/pull/24290#pullrequestreview-464724722.

## How has this been tested?
Confirm all checks pass on this PR.

## Types of changes
Non-breaking change

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/accessibility-coding-standards/ -->
- [x] My code has proper inline documentation. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/ -->
- [x] I've included developer documentation if appropriate. <!-- Handbook: https://developer.wordpress.org/block-editor/ -->
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR. <!-- React Native mobile Gutenberg guidelines: https://github.com/WordPress/gutenberg/blob/master/docs/contributors/native-mobile.md -->

adamziel committed:
Add test for slashing behavior

adamziel committed:
Lint

adamziel committed:
Update tests

draganescu commented:
@adamziel the new test seems to fail.

noisysocks commented on phpunit/class-rest-sidebars-controller-test.php:
```diff
@@ -378,6 +378,60 @@ public function test_update_item() {
 	}
 
+	/**
+	 *
```
Please write something here.

adamziel committed:
Update tests

adamziel committed:
Replace > with &gt;

adamziel committed:
Remove &gt;

adamziel commented:
The test fails because of ">" character in the input. It should most definitely be addressed, but is not trivial and it also seems out of scope for this PR. I removed this case from here and added another task to the list.

adamziel commented:
Would you mind re-reviewing @draganescu?

TimothyBJacobs commented:
> The test fails because of ">" character in the input. It should most definitely be addressed, but is not trivial and it also seems out of scope for this PR. I removed this case from here and added another task to the list.

Which issue # is that?

adamziel commented:
@TimothyBJacobs I was lazy and added it to the list in this issue to address later this week :) https://github.com/WordPress/gutenberg/issues/24530

TimothyBJacobs commented:
Gotcha!
